packages/astro/src/@types/config.ts

```diff
@@ -41,26 +48,26 @@ export interface AstroConfig {
   /** Options for rendering markdown content */
   markdownOptions?: Partial<AstroMarkdownOptions>;
   /** Options specific to `astro build` */
-  buildOptions: {
+  buildOptions?: {
     /** Your public domain, e.g.: https://my-site.dev/. Used to generate sitemaps and canonical URLs. */
     site?: string;
     /** Generate an automatically-generated sitemap for your build.
      * Default: true
      */
-    sitemap: boolean;
+    sitemap?: boolean;
     /**
      * Control the output file/URL format of each page.
      *   If true, Astro will generate a directory with a nested index.html (ex: "/foo/index.html") for each page.
      *   If false, Astro will generate a matching HTML file (ex: "/foo.html") instead of a directory.
      * Default: true
      */
-    pageDirectoryUrl: boolean;
+    pageDirectoryUrl?: boolean;
```

Comment:
At the risk of bikeshedding, is `pageDirectoryUrl` the most clear name for this? Maybe `outputFormat` with an enum value of `'page/index.html' | 'page.html'`?

Reply:
Open to anything here! FYI Hugo calls this option “uglyurls” :)

Reply:
after thinking about it for a bit, I'd rather do a boolean vs. an enum. I think if you saw this in documentation or your own config, you might think that "page.html" means an actual file location (similar to dist, pages, etc.). 

However, I'm still open to better name suggestions if anyone has any!

Reply:
I'd be slightly into `pageDirectoryUrlFormat` instead of `pageDirectoryUrl`, if everyone's okay with merging and releasing this by end of day (since this technically already just went out with the last release a few minutes ago).

Reply:
I'm fine with that.

Reply:
Done! `pageDirectoryUrl ` is now `pageUrlFormat?: 'file' | 'directory'`